test/outtest2.F90

```diff
@@ -102,7 +104,14 @@ program outtest2
   ! Confirm the "missing" value is still the same value that was set previously via the call to setxmiss.
   IF ( nint(xmiss) .ne. nint(getbmiss()) ) stop 3
 
-  call writsb ( 11 )
+  ! Test cnved4 to cover im8b=.true. case
+  call writsa ( 11, mxbfmg, mgbf, lmgbf )
+  mgbf2 = mgbf
+  call cnved4(mgbf,mxbfmg,mgbf2)
+  if ( iupbs01(mgbf2, 'BEN') .ne. 4 ) stop 4
+  ! Re-converting to BUFR ed. 4 should leave the message unchanged
+  call cnved4(mgbf2,mxbfmg,mgbf)
+  if ( .not. all( mgbf(1:lmgbf) .eq. mgbf2(1:lmgbf) )) stop 5
```

Comment:
I'm also not sure I understand why you're calling cnved4 a second time in line 113?  Unless I'm missing something, it seems to me you should be able to just remove line 113, and I think the test in line 114 should still work.

Reply:
Or is the message that comes out of writsa in the mgbf array initially encoded as edition 3?

Reply:
Correct-- the second pass is to test the logic of "converting" from ed 4 to ed 4.